deployment/helm/edc-controlplane/templates/deployment.yaml

```
@@ -33,6 +34,7 @@ spec:
       {{- end }}
       {{- end }}
       serviceAccountName: {{ include "edc-controlplane.serviceAccountName" . }}
+      automountServiceAccountToken: {{ if .Values.automountServiceAccountToken }}true{{ else }}false{{ end }}
```

This should work, because you set it already to a default value within the `values.yaml`, or not?

```suggestion
      automountServiceAccountToken: {{ .Values.automountServiceAccountToken }}
```

---

problem is that k8s expects this to be a boolean value. When passing non boolean values those would be rendered into the resource file and would break the deployment. This way I decided to explicitly evaluate this variable and render its respective value there. 
* automountServiceAccountToken: true renders `true`
* automountServiceAccountToken: false renders `false`
* automountServiceAccountToken: nil renders `false`
* automountServiceAccountToken: "" renders `false`
* automountServiceAccountToken: 0 renders `false`
* automountServiceAccountToken: 1 renders `true`
* automountServiceAccountToken: anytext renders `true`

But if you prefer I can also commit your sugguestion.

---

ah cool, no fine for me thanks!